### DISCUSSION TITLE
Unexpected image size (orientation)

### DISCUSSION THREAD

**Discussion by printmaps** [ImageMagick/ImageMagick#4478] [Help] [answered]
MacOS preview shows the size of my jpg image correct as 2268x4032:

<img width="888" alt="Bildschirmfoto 2021-11-13 um 12 02 56" src="https://user-images.githubusercontent.com/26689799/141630426-2838004c-1a03-431c-aead-afadc891f61c.png">

But IM shows this confusing size:

```
% magick identify ./bilder/20210922_112158.jpg
./bilder/20210922_112158.jpg JPEG 4032x2268 4032x2268+0+0 8-bit sRGB 3.65144MiB 0.000u 0:00.001
```

A format conversion from jpg to png ...

`magick ./bilder/20210922_112158.jpg 20210922_112158.png
`

... leads to:

<img width="865" alt="Bildschirmfoto 2021-11-13 um 12 06 19" src="https://user-images.githubusercontent.com/26689799/141633906-5be53d71-2a11-4e84-8932-bfb67669de4b.png">

And IM shows this:

```
% magick identify 20210922_112158.png           
20210922_112158.png PNG 4032x2268 4032x2268+0+0 8-bit sRGB 15.3445MiB 0.000u 0:00.000
```
That's the IM version I use:

 ```
% magick -version
Version: ImageMagick 7.1.0-13 Q16-HDRI arm 2021-10-29 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(5.0) 
Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib
Compiler: gcc (4.2)
```

**Comment by snibgo** [comment]
I suppose the image within the file actually has 4032 columns and 2268 rows, but also carries metadata that says the image needs to be rotated by -90 degrees.

Some software will automatically apply this metadata. IM won't, unless you specify "-auto-orient" after reading the input.

**Comment by printmaps** [comment, marked as answer]
> I suppose the image within the file actually has 4032 columns and 2268 rows, but also carries metadata that says the image needs to be rotated by -90 degrees.

The metadata for the jpg image show the value '6' for the orientation.

![20190731231914](https://user-images.githubusercontent.com/26689799/141670137-357777b6-584f-4e4e-8189-d7d54cba5808.png)
(Reference: https://jdhao.github.io/2019/07/31/image_rotation_exif_info/)

The option "-auto-orient" solves the problem. Thanks for your the help.
